app/pastes/pasteRepository.js

```diff
@@ -16,6 +17,22 @@ class PasteRepository {
     }
 
     async findAll({ ids, name, authorId }) {
+        if (await this.cacheProvider.exists(`${ids}_${name}_${authorId}`)) {
+            const cachedData = await this.cacheProvider.get(`${ids}_${name}_${authorId}`);
+            const result = cachedData.map(x => this.pasteFactory.create({
```

Comment:
Create a separated "private" method `_buildPasteEntity(row) { return  this.pasteFactory.create({....)` to reduce duplicated code.